common/src/main/java/org/apache/uniffle/common/provider/HadoopAccessorProvider.java

```diff
@@ -0,0 +1,260 @@
+      String msg = String.format("There is need to be interactive with secured DFS by user: %s, path: %s "
+          + "but the HadoopAccessProvider's kerberos config is disabled and can't retrieve the "
+          + "secured filesystem", user, path);
+      throw new Exception(msg);
+    }
+
+    // For local file systems, return the raw local file system, such calls to flush()
+    // actually flushes the stream.
+    try {
+      FileSystem fs;
+      if (retrievedByProxyUser) {
+        LOGGER.info("Fetching the proxy user ugi of {} when getting filesystem of [{}]", user, path);
+        UserGroupInformation proxyUserUGI = provider.getProxyUser(user);
+        fs = proxyUserUGI.doAs(new PrivilegedExceptionAction<FileSystem>() {
+          @Override
+          public FileSystem run() throws Exception {
+            return path.getFileSystem(configuration);
+          }
+        });
+      } else {
+        fs = path.getFileSystem(configuration);
+      }
+      if (fs instanceof LocalFileSystem) {
+        LOGGER.debug("{} is local file system", path);
+        return ((LocalFileSystem) fs).getRawFileSystem();
+      }
+      return fs;
+    } catch (IOException e) {
+      LOGGER.error("Fail to get filesystem of {}", path);
+      throw e;
+    }
+  }
+
+  /**
+   * The only entrypoint is to get the hadoop filesystem instance and is compatible with
+   * the kerberos security.
+   *
+   * When to invoke this method?
+   * 1. For shuffle server side, it needs to get filesystem before writing the shuffle data to secured HDFS
+   * with the spark job's user auth.
+   *
+   * @param user
+   * @param path
+   * @param conf
+   * @return
+   * @throws Exception
+   */
+  public static FileSystem getFilesystem(
+      final String user,
+      final Path path,
+      final Configuration conf) throws Exception {
+    UserGroupInformation.AuthenticationMethod authenticationMethod =
+        SecurityUtil.getAuthenticationMethod(conf);
+    boolean securityEnable = authenticationMethod != UserGroupInformation.AuthenticationMethod.SIMPLE;
+    return getInternalFileSystem(user, securityEnable, path, conf);
+  }
+
+  /**
+   * The method is to return the Hadoop Filesystem directly which is not retrieved by
+   * ugi proxy user.
+   *
+   * When to invoke this method?
+   * 1. In client side, spark shuffle-reader getting filesystem before reading shuffle data stored in HDFS.
+   * 2. In shuffle-server/coordinator side, it reads the config file stored in HDFS.
+   *
+   * @param path
+   * @param configuration
+   * @return
+   * @throws Exception
+   */
+  public static FileSystem getFileSystem(
+      final Path path,
+      final Configuration configuration) throws Exception {
+    return getInternalFileSystem(null, false, path, configuration);
+  }
+
+  /**
+   * For kerberized cluster access
+   * @param securityInfo
+   * @throws Exception
+   */
+  public static void init(SecurityInfo securityInfo) throws Exception {
+    if (provider == null) {
```

Comment:
If this is singleton, could we use lazy holder to keep consistent code style with other code?

Reply:
LazyHolder only works with non-args constructors.